packages/vscode-dependencies-validation/src/autofix/packageJsonFileWatcher.ts

```diff
@@ -0,0 +1,30 @@
+import type { Uri } from "vscode";
+import { debounce } from "lodash";
+import { VscodeFileEventConfig } from "../vscodeTypes";
+import { handlePackageJsonEvent } from "./eventUtil";
+
+const debouncedHandlePackageJsonEvent = debounce(handlePackageJsonEvent, 3000);
+
+// TODO: what should happen after git clone ??
+export function addPackageJsonFileWatcher(
+  vscodeConfig: VscodeFileEventConfig
+): void {
+  const fileWatcher =
+    vscodeConfig.workspace.createFileSystemWatcher("**/package.json");
+
+  fileWatcher.onDidChange(onChange(vscodeConfig));
+  fileWatcher.onDidCreate(onCreate(vscodeConfig));
+}
+
+function onCreate(vscodeConfig: VscodeFileEventConfig): any {
+  return (uri: Uri) => handlePackageJsonEvent(uri, vscodeConfig);
```

Comment:
I think that if a file is created and immediately changed you may end up running
two heavy operations in parallel, it may be safer to always debounce with `debouncedHandlePackageJsonEvent`

Reply:
Done